controllers/eks_controller.go

```diff
@@ -0,0 +1,196 @@
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsmanagedcluster/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsmanagedcluster/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *EKSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var err error
+	logger := r.Log.WithValues("namespace", req.Namespace, "cluster", req.Name)
+
+	logger.Info("Reconciling AWSManagedCluster for EKS")
+
+	cluster := &capa.AWSManagedCluster{}
+	if err := r.Get(ctx, req.NamespacedName, cluster); err != nil {
+		return ctrl.Result{}, microerror.Mask(client.IgnoreNotFound(err))
+	}
+
+	eksCluster := &eks.AWSManagedControlPlane{}
+	if err := r.Get(ctx, req.NamespacedName, eksCluster); err != nil {
+		return ctrl.Result{}, microerror.Mask(client.IgnoreNotFound(err))
+	}
+
+	awsClusterRoleIdentity := &capa.AWSClusterRoleIdentity{}
+	err = r.Get(ctx, types.NamespacedName{Name: eksCluster.Spec.IdentityRef.Name}, awsClusterRoleIdentity)
+	if err != nil {
+		return ctrl.Result{}, microerror.Mask(fmt.Errorf("failed to get AWSClusterRoleIdentity object %q: %w", eksCluster.Spec.IdentityRef.Name, err))
+	}
+
+	arn := awsClusterRoleIdentity.Spec.RoleArn
+
+	// extract AccountID from ARN
+	re := regexp.MustCompile(`[-]?\d[\d,]*[\.]?[\d{2}]*`)
+	accountID := re.FindAllString(arn, 1)[0]
+
+	if accountID == "" {
+		logger.Error(err, "Unable to extract Account ID from ARN")
+		return ctrl.Result{}, microerror.Mask(fmt.Errorf("Unable to extract Account ID from ARN %s", string(arn)))
+	}
+
+	// create the cluster scope.
+	clusterScope, err := scope.NewClusterScope(scope.ClusterScopeParams{
+		AccountID:        accountID,
+		ARN:              arn,
+		BucketName:       key.BucketName(accountID, cluster.Name),
+		ClusterName:      cluster.Name,
+		ClusterNamespace: cluster.Namespace,
+		ConfigName:       key.ConfigName(cluster.Name),
+		Installation:     r.Installation,
+		Region:           eksCluster.Spec.Region,
+		// This is a hack to allow CAPI clusters to drop the 'release.giantswarm.io/version' label.
+		ReleaseVersion: "20.0.0-alpha1",
+		SecretName:     key.SecretName(cluster.Name),
+
+		Logger:  logger,
+		Cluster: cluster,
+	})
+	if err != nil {
+		return reconcile.Result{}, microerror.Mask(err)
+	}
+
+	// Create IRSA service.
+	irsaService := irsaEks.New(clusterScope, r.Client)
+
+	if cluster.DeletionTimestamp != nil {
+		finalizers := cluster.GetFinalizers()
+		if !key.ContainsFinalizer(finalizers, key.FinalizerName) && !key.ContainsFinalizer(finalizers, key.FinalizerNameDeprecated) {
```

Comment:
There is no need to check for the deprecated finalizer if we are reconciling CRs that were never reconciled by older versions of the controller.